taskcluster/fenix_taskgraph/transforms/multi_dep.py

```diff
@@ -23,7 +23,8 @@ def build_name_and_attributes(config, tasks):
         }
         primary_dep = task["primary-dependency"]
         copy_of_attributes = primary_dep.attributes.copy()
-        task.setdefault("attributes", {}).update(copy_of_attributes)
+        copy_of_attributes.update(task.get("attributes", {}))
+        task["attributes"] = copy_of_attributes
```

Comment:
I'd rename `copy_of_attributes` to `attributes` since it's no longer just a copy.

Reply:
Good call, done!